packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -0,0 +1,233 @@
+        );
+
+      setExpandedSidebarItemIds(initialExpandedSidebarItemIds);
+    }
+  }, [depth, subNavigation]);
+
+  const handleSidebarItemClick = React.useCallback(
+    (itemId: string) => () => {
+      setExpandedSidebarItemIds((previousValue) =>
+        previousValue.includes(itemId)
+          ? previousValue.filter((previousValueItemId) => previousValueItemId !== itemId)
+          : [...previousValue, itemId],
+      );
+    },
+    [],
+  );
+
+  return (
+    <List sx={{ mb: depth === 0 ? 4 : 1, pl: 2 * depth }}>
+      {subNavigation.map((navigationItem, navigationItemIndex) => {
+        if (navigationItem.kind === 'header') {
+          return (
+            <ListSubheader key={`subheader-${depth}-${navigationItemIndex}`} component="div">
+              {navigationItem.title}
+            </ListSubheader>
+          );
+        }
+
+        if (navigationItem.kind === 'divider') {
+          const nextItem = subNavigation[navigationItemIndex + 1];
+
+          return (
+            <Divider
+              key={`divider-${depth}-${navigationItemIndex}`}
+              sx={{ mt: 1, mb: nextItem.kind === 'header' ? 0 : 1 }}
+            />
+          );
+        }
+
+        const navigationItemId = `${navigationItem.title}-${depth}-${navigationItemIndex}`;
+
+        const isNestedNavigationExpanded = expandedSidebarItemIds.includes(navigationItemId);
+
+        const nestedNavigationCollapseIcon = isNestedNavigationExpanded ? (
+          <ExpandLessIcon />
+        ) : (
+          <ExpandMoreIcon />
+        );
+
+        const listItem = (
+          <ListItem>
+            <ListItemButton
+              selected={IS_CLIENT && hasMounted && window.location.pathname === navigationItem.path}
+              onClick={handleSidebarItemClick(navigationItemId)}
+            >
+              <ListItemIcon>{navigationItem.icon}</ListItemIcon>
+              <ListItemText primary={navigationItem.title} />
+              {navigationItem.children ? nestedNavigationCollapseIcon : null}
+            </ListItemButton>
+          </ListItem>
+        );
+
+        return (
+          <React.Fragment key={navigationItemId}>
+            {navigationItem.path ? (
+              <a href={navigationItem.path} style={{ color: 'inherit', textDecoration: 'none' }}>
+                {listItem}
+              </a>
+            ) : (
+              listItem
+            )}
+            {IS_CLIENT && hasMounted && navigationItem.children ? (
+              <Collapse in={isNestedNavigationExpanded} timeout="auto" unmountOnExit>
+                <DashboardSidebarSubNavigation
+                  subNavigation={navigationItem.children}
+                  depth={depth + 1}
+                />
+              </Collapse>
+            ) : null}
+          </React.Fragment>
+        );
+      })}
+    </List>
+  );
+}
+
+interface DashboardLayoutProps {
+  children: React.ReactNode;
+  sidebarWidth?: number;
```

Comment:
Does this really have to be configurable? And if it were configurable, wouldn't it make sense to be responsive? What if we just come up with good defaults based on the breakpoint and wait for making it configurable on user feedback?

Does the drawer collapse on narrower screens?

Reply:
It was just a nice easy to add option, didn't look into responsiveness though.

> Does the drawer collapse on narrower screens?

No.